### PR TITLE
Move oembed to src

### PR DESCRIPTION
**Pull request by MrPetovan**
Related to #4155

Part of #3878
Part of #4090

This is just a preparation for the work that hopefully will be agreed upon in #4155

### CODE REVIEW

File: src/Content/OEmbed.php [hunk 1]
```diff
@@ -0,0 +1,352 @@
+<?php
+
+/*
+ * To change this license header, choose License Headers in Project Properties.
```
Comment (annando) [inline src/Content/OEmbed.php]: These lines confusing me. Are these place holders?

Additionally ```@file``` is missing.

File: src/Content/OEmbed.php [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+use dba;
+use DOMDocument;
+use DOMXPath;
+use DOMNode;
+
+require_once 'include/dba.php';
+require_once 'mod/proxy.php';
+
+/**
+ * Description of OEmbed
```
Comment (annando) [inline src/Content/OEmbed.php]: A little bit too generic, I think.

File: src/Content/OEmbed.php [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+ *
+ * @author benlo
+ */
+class OEmbed
+{
+	public static function replaceCallback($matches)
+	{
+		$embedurl = $matches[1];
+		$j = OEmbed::fetchURL($embedurl);
```
Comment (annando) [inline src/Content/OEmbed.php]: Wouldn't ```self::``` be better here? (Same goes for the line below and much other places)

File: src/Content/OEmbed.php [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+	 * @brief Generates attribute search XPath string
+	 *
+	 * @param string $attr Name of the attribute to seach
+	 * @param string $value Value to search in a space-separated list
+	 * @return string
+	 */
+	private static function buildXPath($attr, $value)
+	{
+		// https://www.westhoffswelt.de/blog/2009/6/9/select-html-elements-with-more-than-one-css-class-using-xpath
+		return "contains( normalize-space( @$attr ), ' $value ' ) or substring( normalize-space( @$attr ), 1, string-length( '$value' ) + 1 ) = '$value ' or substring( normalize-space( @$attr ), string-length( @$attr ) - string-length( '$value' ) ) = ' $value' or @$attr = '$value'";
```
Comment (annando) [inline src/Content/OEmbed.php]: There are several spaces in that line that could be removed to make it look better.

Comment (MrPetovan) [reply src/Content/OEmbed.php]: Some spaces are required are it is looking for a value in a space-separated list, but there are a several others I can indeed remove.

**Pull request merged**